__tests__/components/MosaicAmountDisplay.spec.ts

```diff
@@ -0,0 +1,355 @@
+            // Arrange:
+            const id = NamespaceId.createFromEncoded(mockNetworkCurrency.namespaceIdHex);
+            const networkCurrency = mockNetworkCurrency;
+            const expectedResult = true;
+
+            // Act + Assert:
+            runUseNetworkTest(id, networkCurrency, expectedResult);
+        });
+
+        test('return false when id is not belong to network currency', () => {
+            // Arrange:
+            const id = NamespaceId.createFromEncoded('E84B99BA41F4AFEE');
+            const networkCurrency = mockNetworkCurrency;
+            const expectedResult = false;
+
+            // Act + Assert:
+            runUseNetworkTest(id, networkCurrency, expectedResult);
+        });
+
+        test('return false when network currency is not loaded', () => {
+            // Arrange:
+            const id = NamespaceId.createFromEncoded('E84B99BA41F4AFEE');
+            const networkCurrency = null;
+            const expectedResult = false;
+
+            // Act + Assert:
+            runUseNetworkTest(id, networkCurrency, expectedResult);
+        });
+    });
+
+    describe('get divisibility()', () => {
+        const runGetDivisibilityTest = (id: MosaicId, useNetwork: boolean, expectedResult: number) => {
+            // Arrange:
+            const props = {
+                id,
+            };
+            const mockUseNetwork = jest.fn().mockReturnValue(useNetwork);
+
+            // Act:
+            const wrapper = getMosaicAmountDisplayWrapper(props);
+            const component = wrapper.vm as MosaicAmountDisplayTs;
+            component['useNetwork'] = mockUseNetwork;
+            const result = component['divisibility'];
+
+            // Assert:
+            expect(result).toBe(expectedResult);
+        };
+
+        test('return network currency divisibility when useNetwork() returns true', () => {
+            // Arrange:
+            const id = null;
+            const useNetwork = true;
+            const expectedResult = mockNetworkCurrency.divisibility;
+
+            // Act + Assert:
+            runGetDivisibilityTest(id, useNetwork, expectedResult);
+        });
+
+        test('return mosaic divisibility for given id when useNetwork() returns false', () => {
+            // Arrange:
+            const mosaic = mosaicsMock[1];
+            const id = new MosaicId(mosaic.mosaicIdHex);
+            const useNetwork = false;
+            const expectedResult = mosaic.divisibility;
+
+            // Act + Assert:
+            runGetDivisibilityTest(id, useNetwork, expectedResult);
+        });
+
+        test('return divisibility from config for unknown mosaic when useNetwork() returns false', () => {
+            // Arrange:
+            const id = new MosaicId('100D8D4D98285200');
+            const useNetwork = false;
+            const expectedResult = 100;
+
+            // Act + Assert:
+            runGetDivisibilityTest(id, useNetwork, expectedResult);
```

Comment:
in what situation can this happen where the mosaic being displayed is unknown and this fallback makes sense?

Reply:
As I can see, for some reason wallet gets the `divisibility` value from owned mosaics. So when transaction contains mosaic which is not owned by account it will fallback and get divisibility from config (what is totally wrong 😬).

Fortunately, the `divisibility` is missing in the config and the wrong amount is not displayed. This needs to be fixed. Opened an issue: https://github.com/symbol/desktop-wallet/issues/1908

Reply:
I think it can be fixed in a separate PR.